File: demo1/src/main/java/com/example/demo1/base/BaseFragment.java
package com.example.demo1.base;

import android.app.Activity;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.widget.Toolbar;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.example.demo1.R;

import me.yokeyword.fragmentation.SupportFragment;
import me.yokeyword.fragmentation_swipeback.SwipeBackFragment;

/**
 * Created by yida on 2018/12/26.
 */
public abstract class BaseFragment extends SwipeBackFragment {
    public Activity mActivity;
    protected View mView;

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        mView = inflater.inflate(getLayoutId(), null);
        return isSupportSwipeBack() ? attachToSwipeBack(mView) : mView;
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        mActivity = getActivity();
        initView(mView);
    }

    public void initToolbar(View view, String title) {
        Toolbar toolbar = view.findViewById(R.id.toolbar);
        TextView textView = toolbar.findViewById(R.id.toolbar_title);
        if (toolbar.findViewById(R.id.toolbar_title) != null) {
//            toolbar.findViewById(R.id.toolbar_title).setText();
            textView.setText(title);
        } else {
            toolbar.setTitle(title);
        }
    }

    /**
     * 处理物理返回按键
     * @return
     */
    @Override
    public boolean onBackPressedSupport() {
        pop();
        return true;
    }

    protected boolean isSupportSwipeBack() {
        return false;
    }

    protected abstract void initView(View view);

    protected abstract int getLayoutId();


}

File: kaiyanapp/src/main/java/com/example/yida/kaiyanapp/base/BaseFragment.kt
package com.example.yida.kaiyanapp.base

import android.os.Bundle
import android.support.annotation.LayoutRes
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.example.yida.kaiyanapp.MyApplication
import com.example.yida.kaiyanapp.view.MultipleStatusView

/**
 * Created by yida on 2018/8/30.
 */
abstract class BaseFragment: Fragment() {

    /**
     * 视图是否加载完毕
     */
    private var isViewPrepars = false

    /**
     * 数据是否加载过了
     */
    private var HasLoadData = false

    /**
     * 多种状态的 view 的切换
     */
    protected var mLayoutStatusView: MultipleStatusView? = null

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View {
        return inflater!!.inflate(getLayoutId(), null)
//        return inflater.inflate(getLayoutId(),null)
    }

    override fun setUserVisibleHint(isVisibleToUser: Boolean) {
        super.setUserVisibleHint(isVisibleToUser)
        if (isVisibleToUser) {
            lazyLoadDataIfPrepared()
        }
    }

    private fun lazyLoadDataIfPrepared() {
        if (userVisibleHint && isViewPrepars && !HasLoadData) {
            lazyLoad()
            HasLoadData = true
        }
    }

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        isViewPrepars = true
        initView()
        lazyLoadDataIfPrepared()
        // 多种状态切换的view 重试点击事件
        mLayoutStatusView?.setOnRetryClickListener(mRetryClickListener)
    }

    open val mRetryClickListener: View.OnClickListener = View.OnClickListener {
        lazyLoad()
    }

    /**
     * 加载布局
     */
    @LayoutRes
    abstract fun getLayoutId():Int

    /**
     * 初始化 ViewI
     */
    abstract fun initView()

    /**
     * 懒加载
     */
    abstract fun lazyLoad()

    override fun onDestroy() {
        super.onDestroy()
        activity?.let { MyApplication.getRefWatcher(it)?.watch(activity) }
    }
}